e2e/src/test/java/io/aiven/kafka/tieredstorage/e2e/SingleBrokerTest.java

```diff
@@ -525,11 +584,18 @@ void remoteCleanupDueToRetention() throws Exception {
             // TODO check for EARLIEST_LOCAL_TIMESTAMP when available in client
 
             consumer.assign(List.of(TP_0_0));
-            consumer.seek(TP_0_0, 0);
-
-            final ConsumerRecord<byte[], byte[]> record = consumer.poll(Duration.ofSeconds(1)).records(TP_0_0).get(0);
-            assertThat(record.offset()).isEqualTo(newStartOffset);
-
+            consumer.seekToBeginning(List.of(TP_0_0));
+
+            // with current retention logic, active segment is rotated, and no data is available locally.
+            final List<ConsumerRecord<byte[], byte[]>> records = consumer.poll(Duration.ofSeconds(1)).records(TP_0_0);
+            assertThat(records).isEmpty();
```

Comment:
Main change: validating that the active segment is not rotated is not correct anymore.